#	
#	File to drive the testing of the config file class
#	testGoal::config file::expected line Number of error (0) is success::expected error message
#
#	the loader expects a file with the same name as the config file ".py" to contain a
#	method called checkMethod.  This takes no parameters as config is global

Test 1 - Check Empty File::configfile_test1::config_file_not_called.py::0::Missing key -> 'LOG_DIRECTORY'

Test 2 - Not Accessible::configfile_test-not-there::config_file_not_called.py::0::Error accessing config file -> '[Errno 2] No such file or directory: '/Users/stu/Development/scripts/dtpo_autoload/tests/configFileTests/configfile_test-not-there''

Test 3 - Innaccessible Source Directory::noaccess/configfile_test3::config_file_not_called.py::0::Error accessing config file -> '[Errno 13] Permission denied: '/Users/stu/Development/scripts/dtpo_autoload/tests/configFileTests/noaccess/configfile_test3''

Test 4 - Missing Attributes::configfile_test4::config_file_not_called.py::37::Missing key -> 'ORPHAN_DOCUMENTS_DIRECTORY'

Test 5 - Duplicated Attributes::configfile_test5::config_file_not_called.py::43::Duplicate key -> 'PATTERN_FILE', now -> 'Duplicated Entry', was -> '/Users/stu/Development/scripts/dtpo_autoload/dtpo_autoload_patterns.conf''

Test 6 - Attributes with missing value (spaces)::configfile_test6::config_file_not_called.py::11::Missing value for key -> 'LOG_DIRECTORY'

Test 7 - Atributes with missing key::configfile_test7::config_file_not_called.py::39::Missing key -> 'LOG_DIRECTORY'

Test 8 - Good File with no Syntax variation::configfile_test8::test8.py::0::???

Test 9 - Good file with offset comments etc::configfile_test9::test8.py::0::???

Test 10 - Catch DEBUG enabled before Log Directory Set::configfile_test10::config_file_not_called.py::9::DEBUG enabled without LOG_DIRECTORY set

Test 11 - Good file with inaccessible log directory::configfile_test11::config_file_not_called.py::10::Log directory not accessible -> '/Users/stu/Development/scripts/dtpo_autoload/tests/configFileTests/noaccess', error -> directory not accessible. Error -> '[Errno 13] Permission denied: '/Users/stu/Development/scripts/dtpo_autoload/tests/configFileTests/noaccess/testFile.Remove''

Test 12 - Attributes with missing value (no spaces)::configfile_test12::config_file_not_called.py::39::Missing key -> 'LOG_DIRECTORY'

Test 13 - Good File but innaccessible working directory::configfile_test13::test13.py::44::SOURCE_DIRECTORY not accessible -> directory not accessible. Error -> '[Errno 13] Permission denied: '/Users/stu/Development/scripts/dtpo_autoload/tests/configFileTests/noaccess/testFile.Remove''

Test 14 - Pattern File not accessible::configfile_test14::test14.py::44::File PATTERN_FILE not accessible -> [Errno 13] Permission denied: '/Users/stu/Development/scripts/dtpo_autoload/tests/configFileTests/noaccess'
